loony.nim

```diff
@@ -309,9 +310,15 @@ proc popImpl[T](queue: LoonyQueue[T]; forcedCoherance: static bool = false): T =
           # cache lines with atomic writes and loads rather than requiring a
           # CPU to completely commit its STOREBUFFER
 
+          # when T is ref:
+          #   var res = cast[Loonatic[T]](prev and SLOTMASK)
+          #   result = extract res
+          # else:
           result = cast[T](prev and SLOTMASK)
-          assert not result.isNil
-          when result is ref:
+          when T is ref or T is pointer:
+            assert not result.isNil
```

Comment:
If you're going to potentially raise a defect in someone else's code, please add a string argument explaining to the user what the problem is.  I would prefer to just remove it and cover misbehavior with a unit test.

Is it likely someone passed a `T` that is not `ref` or `pointer`?

Reply:
I originally had an opposing assertion which would assert that non-pointers would be compared to a default of their type.